### PR TITLE
Feature/put object iodata

### PR DESCRIPTION
**Pull request by karlseguin**
Addresses #305  but...

There didn't seem to be any test coverage for the current implementation and, when in Rome, ... I did manually test it.

Also, as far as I can tell, the `no_encode: true` option is necessary to maintain backwards compatibility.

### CODE REVIEW

**Comment by benwilson512** [comment]
Thank you! Merging to a local branch for testing.

File: lib/ex_aws/operation/s3.ex [hunk 1]
```diff
@@ -34,7 +34,7 @@ defmodule ExAws.Operation.S3 do
 
       headers = headers
       |> Map.put("x-amz-content-sha256", hashed_payload)
-      |> Map.put("content-length", byte_size(body))
+      |> Map.put("content-length", :erlang.iolist_size(body))
```
Comment (binarylogic) [inline lib/ex_aws/operation/s3.ex]: Won't this break backwards compatibility? We resolved this via a [custom `get_byte_size` function](https://github.com/timberio/ex_aws/blob/master/lib/ex_aws/operation/s3.ex#L59-L65)

Comment (benwilson512) [reply lib/ex_aws/operation/s3.ex]:
```
iex(1)> :erlang.iolist_size("")
0
```
seems backwards compatible to me, do you have an example where it breaks?

Comment (binarylogic) [reply lib/ex_aws/operation/s3.ex]: Ah, yes, you're correct. I assumed it accepted a list only.

**Pull request merged**